src/application.rs

```diff
@@ -18,69 +18,80 @@ pub trait ApplicationHandler {
+    /// See [`destroy_surfaces()`].
     ///
-    /// On Web, the `Resumed` event is emitted in response to a [`pageshow`] event
-    /// with the property [`persisted`] being true, which means that the page is being
-    /// restored from the [`bfcache`] (back/forward cache) - an in-memory cache that
-    /// stores a complete snapshot of a page (including the JavaScript heap) as the
-    /// user is navigating away.
+    /// ## Portability
     ///
-    /// [`pageshow`]: https://developer.mozilla.org/en-US/docs/Web/API/Window/pageshow_event
-    /// [`persisted`]: https://developer.mozilla.org/en-US/docs/Web/API/PageTransitionEvent/persisted
-    /// [`bfcache`]: https://web.dev/bfcache/
-    /// [`Suspended`]: Self::suspended
-    fn resumed(&mut self, event_loop: &ActiveEventLoop);
+    /// It's recommended that applications should only initialize their render surfaces after the
+    /// [`create_surfaces()`] method is called. Some systems (specifically Android) won't allow
+    /// applications to create a render surface until that point.
+    ///
+    /// For consistency, all platforms call this method even if they don't themselves have a formal
+    /// surface destroy/create lifecycle. For systems without a surface destroy/create lifecycle the
+    /// [`create_surfaces()`] event is always emitted after the [`StartCause::Init`] event.
+    ///
+    /// Applications should be able to gracefully handle back-to-back [`create_surfaces()`] and
+    /// [`destroy_surfaces()`] calls.
+    ///
+    /// ## Platform-specific
+    ///
+    /// ### Android
+    ///
+    /// On Android, the [`create_surfaces()`] method is called when a new [`SurfaceView`] has been
+    /// created. This is expected to closely correlate with the [`onResume`] lifecycle event but
+    /// there may technically be a discrepancy.
+    ///
+    /// [`onResume`]: https://developer.android.com/reference/android/app/Activity#onResume()
+    ///
+    /// Applications that need to run on Android must wait until they have been "resumed" before
+    /// they will be able to create a render surface (such as an `EGLSurface`, [`VkSurfaceKHR`]
+    /// or [`wgpu::Surface`]) which depend on having a [`SurfaceView`]. Applications must also
+    /// assume that if they are [suspended], then their render surfaces are invalid and should
+    /// be dropped.
+    ///
+    /// [suspended]: Self::destroy_surfaces
+    /// [`SurfaceView`]: https://developer.android.com/reference/android/view/SurfaceView
+    /// [Activity lifecycle]: https://developer.android.com/guide/components/activities/activity-lifecycle
+    /// [`VkSurfaceKHR`]: https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkSurfaceKHR.html
+    /// [`wgpu::Surface`]: https://docs.rs/wgpu/latest/wgpu/struct.Surface.html
+    ///
+    /// [`create_surfaces()`]: Self::create_surfaces
+    /// [`destroy_surfaces()`]: Self::destroy_surfaces
+    fn create_surfaces(&mut self, event_loop: &ActiveEventLoop);
```

Comment:
I'm not sure I entirely like this method, but the alternative is to store windows in winit and indicate the creation with `WindowId`.

Though, unless the event loop itself could be less backend agnostic I don't think it'll happen.

Reply:
This isn't about `Window` anymore, but only about `RawWindowHandle`.
The hope is that in the future we won't need this anymore if #3695 is resolved, but yeah, otherwise I can't come up with a better idea.